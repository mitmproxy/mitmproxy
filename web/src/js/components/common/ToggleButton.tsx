import React from "react";

type ToggleButtonProps = {
    checked: boolean;
    onToggle: () => any;
    text: string;
};

export default function ToggleButton({
    checked,
    onToggle,
    text,
}: ToggleButtonProps) {
    return (
        <div
            className={
                "btn btn-toggle " + (checked ? "btn-primary" : "btn-default")
            }
            onClick={onToggle}
        >
            <i
                className={
                    "fa fa-fw " +
                    (checked ? "fa-check-square-o" : "fa-square-o")
                }
            />
            &nbsp;
            {text}
        </div>
    );
}
