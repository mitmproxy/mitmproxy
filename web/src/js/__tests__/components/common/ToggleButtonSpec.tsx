import * as React from "react";
import renderer from "react-test-renderer";
import ToggleButton from "../../../components/common/ToggleButton";

describe("ToggleButton Component", () => {
    const mockFunc = jest.fn();

    it("should render correctly", () => {
        const checkedButton = renderer.create(
            <ToggleButton checked={true} onToggle={mockFunc} text="foo" />,
        );
        const tree = checkedButton.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should handle click action", () => {
        const uncheckButton = renderer.create(
            <ToggleButton checked={false} onToggle={mockFunc} text="foo" />,
        );
        const tree = uncheckButton.toJSON();
        tree.props.onClick();
        expect(mockFunc).toBeCalled();
    });
});
