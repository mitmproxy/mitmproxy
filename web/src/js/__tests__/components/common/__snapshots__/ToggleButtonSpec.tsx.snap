// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton Component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="btn btn-toggle btn-primary"
  >
    <i
      class="fa fa-fw fa-check-square-o"
    />
    Â foo
  </div>
</DocumentFragment>
`;
